File: MovieTutorial/MovieTutorial.Web/Modules/MovieDB/Movie/MovieForm.cs
using MovieTutorial.Web.Modules.MovieDB.Movie;
using Serenity.ComponentModel;
using System;
using System.Collections.Generic;

namespace MovieTutorial.MovieDB.Forms;

[FormScript("MovieDB.Movie")]
[BasedOnRow(typeof(MovieRow), CheckNames = true)]
public class MovieForm
{
    public string Title { get; set; }

    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }
    [MovieCastEditor, IgnoreName]
    public List<MovieCastRow> CastList { get; set; }
    public string PrimaryImage { get; set; }
    public string GalleryImages { get; set; }
    [TextAreaEditor(Rows = 8)]
    public string Storyline { get; set; }
    public int Year { get; set; }
    public DateTime ReleaseDate { get; set; }
    public int Runtime { get; set; }
    public List<int> GenreList { get; set; }

    public MovieKind Kind { get; set; }
   

}

File: MovieTutorial/MovieTutorial.Web/Modules/ServerTypes/MovieDB/MovieForm.ts
﻿import { StringEditor, TextAreaEditor, IntegerEditor, DateEditor, LookupEditor, EnumEditor, PrefixedContext } from "@serenity-is/corelib";
import { MovieCastEditor } from "@/MovieDB/MovieCast/MovieCastEditor";
import { MovieKind } from "../Web/Modules.MovieDB.Movie.MovieKind";
import { initFormType } from "@serenity-is/corelib/q";

export interface MovieForm {
    Title: StringEditor;
    Description: TextAreaEditor;
    CastList: MovieCastEditor;
    Storyline: TextAreaEditor;
    Year: IntegerEditor;
    ReleaseDate: DateEditor;
    Runtime: IntegerEditor;
    GenreList: LookupEditor;
    Kind: EnumEditor;
}

export class MovieForm extends PrefixedContext {
    static formKey = 'MovieDB.Movie';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MovieForm.init)  {
            MovieForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = MovieCastEditor;
            var w3 = IntegerEditor;
            var w4 = DateEditor;
            var w5 = LookupEditor;
            var w6 = EnumEditor;

            initFormType(MovieForm, [
                'Title', w0,
                'Description', w1,
                'CastList', w2,
                'Storyline', w1,
                'Year', w3,
                'ReleaseDate', w4,
                'Runtime', w3,
                'GenreList', w5,
                'Kind', w6
            ]);
        }
    }
}

[MovieKind]; // referenced types